internal/controllers/machineset/machineset_controller.go

```diff
@@ -969,12 +994,47 @@ func (r *Reconciler) getMachineNode(ctx context.Context, cluster *clusterv1.Clus
 
 func (r *Reconciler) reconcileUnhealthyMachines(ctx context.Context, cluster *clusterv1.Cluster, ms *clusterv1.MachineSet, filteredMachines []*clusterv1.Machine) (ctrl.Result, error) {
 	log := ctrl.LoggerFrom(ctx)
+
+	// Calculate how many in flight machines we should remediate.
+	// By default, we allow all machines to be remediated at the same time.
+	maxInFlight := len(filteredMachines)
+
+	// If the MachineSet is part of a MachineDeployment, only allow remediations if
+	// it's the desired revision.
+	if util.HasOwner(ms.GetOwnerReferences(), clusterv1.GroupVersion.String(), []string{"MachineDeployment"}) {
```

Comment:
Wondering if we should check for the MD label instead (not sure). We seem to rely on both

Reply:
I'll make a method